sw/device/lib/dif/dif_aes.h

```diff
@@ -101,26 +101,71 @@ typedef enum dif_aes_operation {
 typedef enum dif_aes_mode {
   /**
    * The Electronic Codebook Mode.
+   * In ECB cipher mode the key must be changed for every new block of data.
+   * This is the only secure way to use ECB cipher mode.
+   *
+   * Note: ECB cipher mode don't use the iv parameter of the `dif_aes_start`
+   * function.
+   *
+   * Note: it is discouraged to use this cipher mode, due to impractical amount
+   *       of different keys required to encrypt/decrypt multi-block messages.
    */
   kDifAesModeEcb = 1,
 
   /**
    * The Cipher Block Chaining Mode.
+   *
+   * In CBC cipher mode, the same key can be used for all messages, however
+   * new Initialisation Vector (IV) must be generated for any new message. The
+   * following condition must be true:
+   *     The IV must be unpredictable (it must not be possible to predict the IV
+   *     that will be associated to the plaintext in advance of the generation
+   * of the IV).
```

Comment:
Nit: indentation

Reply:
Thanks for noticing, but I think I'll let as it is, since `clang-format` is not complaining.